test/integration-new/snat/snat_test.go

```diff
@@ -19,6 +19,16 @@ const (
 )
 
 var _ = Describe("SNAT test", func() {
+	It("Pod in private subnet should have Internet access with External SNAT disabled", func() {
+		By("Setting External SNAT to true")
+		k8sUtils.AddEnvVarToDaemonSetAndWaitTillUpdated(f, utils.AwsNodeName, utils.AwsNodeNamespace, utils.AwsNodeName, map[string]string{
+			"AWS_VPC_K8S_CNI_EXTERNALSNAT": "false",
+		})
+
+		By("Checking External Domain Connectivity")
+		ValidateExternalDomainConnectivity(EXTERNAL_DOMAIN)
```

Comment:
Won't this function fail since there won't be external connectivity?

Reply:
The private subnet routing table points to a NAT GW (which resides inside a public subnet). So pods can always init connection. So this doesn't fail.